src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
+            return static::decodeUuid($this->{$uuidKey});
+        }
+
+        return parent::getAttribute($key);
+    }
+
+    public function setAttribute($key, $value)
+    {
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
+            $value = static::encodeUuid($value);
         }
 
-        $data = parent::toArray();
+        return parent::setAttribute($key, $value);
+    }
+
+    protected function getUuidTextAttributeSuffix()
+    {
+        return (property_exists($this, 'uuidTextAttribSuffix')) ? $this->uuidTextAttribSuffix : '_text';
+    }
+
+    protected function uuidTextAttribute($key)
+    {
+        $attribute_s = $this->getKeyName();
+        $suffix = $this->getUuidTextAttributeSuffix();
+        $offset = -(strlen($suffix));
+
+        if (is_string($attribute_s)) {
+            $attribute_s = [$attribute_s];
+        }
 
-        if (isset($data[$this->getKeyName()])) {
-            $data[$this->getKeyName()] = $this->uuid_text;
+        if (substr($key, $offset) == $suffix && in_array(($uuidKey = substr($key, 0, $offset)), $attribute_s)) {
+            return $uuidKey;
         }
 
-        return $data;
+        return false;
+    }
+
+    public function getUuidAttributes()
```

Comment:
The return type here can be `array`, or am I mistaken?

Reply:
return type could be `null` actually (not just `false` - generally anything "falsey") on that line (The return type for the method `uuidTextAttribute` is actually _mixed_).